pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -112,18 +115,17 @@ def advance(self, batch, batch_idx):
         # let logger connector extract current batch size
         self.trainer.logger_connector.on_train_split_start(batch_idx, split_idx, split_batch)
 
+        # choose which loop will run the optimization
         if self.trainer.lightning_module.automatic_optimization:
-            # in automatic optimization, hand over execution to the OptimizerLoop
             optimizers = _get_active_optimizers(self.trainer.optimizers, self.trainer.optimizer_frequencies, batch_idx)
```

Comment:
We can make get_active_optimizers a static_method ?

Reply:
Users might want to override to implement custom scheduling strategy.

Reply:
Ignore it if you want as not fully relevant to this PR.

Reply:
It could be a good idea. I guess it was put in utilities to avoid traversing the loops for it because the `EpochLoop`, `BatchLoop` and `FinetuniningCallback` all use it.

cc @awaelchli 

But definitely not in this PR

Reply:
Yes, as @carmocca said I moved `_get_active_optimizers` out of the loops because it was used in different loops and defined on the TrainingEpochLoop. But if it becomes a method, IMO it should be in the class that owns the optimizers